[BPS-29] : Validator on private handlers

maissanouhaud opened the pull request.

Nysnaa committed:
[*]package.json : added yup package | starting validator

Nysnaa committed:
[WIP][*] added validator and its schema, now have to upgrade it for the optional argument {inventory_address} and the error handling about it.
Also have to finish the unitary test for the validator

podefr commented on src/services/validatorSchema.ts:
```diff
@@ -0,0 +1,6 @@
+import * as yup from 'yup';
+
+export const inventoryCreateSchema = yup.object({
+    user_id: yup.string().defined("All fields must be completed"),
```
same here, let's have an explicit error, for instance "user_id must be provided"

Nysnaa committed:
[WIP][
[*]validatorSchema : changing the validation from public to private handlers

Nysnaa committed:
[NO_PUSH_MAIN] package.json : added script for dev

Nysnaa committed:
[WIP] [*] trying to run test for the validation

podefr commented on src/services/validatorSchema.ts:
```diff
@@ -0,0 +1,16 @@
+import * as yup from 'yup';
+
+export const inventoryItemSchema = yup.object({
+    inventory_address: yup.string().typeError("inventory_address must be a string").defined("The inventory address must be provided, don't forget it ! "),
```
same comments here as in the `item-store` PR:
- Let's make the error message even more explicit and include the expected type for the field.
- let's make the type stricts, for instance if a number is provided let's still have the validation fail even if it could be converted to a string, this would prevent issues later on when we tokenize items

podefr commented on src/services/validatorSchema.ts:
```diff
@@ -0,0 +1,16 @@
+    no_update_after: yup.string().typeError("available_quantity must be a number").optional(),
+    item_id: yup.number().typeError("item_id must be a number").positive("The item_id must be a positive number.").defined("The item_id must be provided, don't forget it ! "),
+    name: yup.string().typeError("name must be a string").defined("The name must be provided, don't forget it ! "),
+    data: yup.object().typeError("data must be an object").defined("The data must be provided, don't forget it ! "),
```
same here, let's be stricter and have `data` be an object that takes pairs of `string` and `string`

podefr commented:
same comments here as well as in the `item-store` PR:

- Can you run prettier to ensure proper formatting?
- Can you squash all your commits into one with a nice title containing the ticket number, and a list of bullet points describing your changes? example of a good commit message: https://github.com/jwa-lab/item-store/commit/b50e3b6c4606c06285064314e2e2ee753085e07d

podefr reviewed (changes requested):
see my comments in the files and the conversation

maissanouhaud committed:
[FIXED][*] add-item.test.js : fixed the indentation

[FIXED][*] add-item.test.js : fixed the indentation

[FIXED][*] add-item.test.js : fixed the indentation

[FIXED][*] add-item.test.js : fixed the indentation

[FIXED][*] add-item.test.js : fixed the indentation

[WIP]
[*]nats.ts / inventory.ts : eslint
[*]validatorSchema : validator more strict for the string
